### PR TITLE
[PHP 8.0] Add SetStateToStaticRector

### PR DESCRIPTION
**Pull request by simivar**
Part of #4765. I didn't use `ChangeMethodVisibilityRector` as it required a class name to be specified and this one should be a generic one.

A way to use generic `ChangeMethodVisibilityRector` could be to refactor it to not require class name but I don't know if that's a visible solution for you.

### CODE REVIEW

**Comment by simivar** [comment]
@TomasVotruba I might need some guidance on how to make it work with PHP 8 tests.

**Comment by TomasVotruba** [comment]
I'd skip the test on PHP 8, as the code is not PHP 8 compatible.

Something like:

```php
@requires PHP < 8
```

File: rules/php80/src/Rector/ClassMethod/SetStateToStaticRector.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            return true;
+        }
+
+        if ($classMethod->isStatic()) {
+            return true;
+        }
+
+        return false;
+    }
+}
```
Comment (TomasVotruba) [inline rules/php80/src/Rector/ClassMethod/SetStateToStaticRector.php:80]: :clap:

**Comment by TomasVotruba** [comment]
> I didn't use ChangeMethodVisibilityRector as it required a class name to be specified and this one should be a generic one.

That's probably a good choice... I wonder, would this doo?

```php
[stdClass::class, '__set_state', 'static']
```

**Comment by TomasVotruba** [comment]
Ah, it doesn't seem to work: https://3v4l.org/7RJ0X

**Comment by TomasVotruba** [comment]
Thank you :+1:

File: rules/php80/src/Rector/ClassMethod/SetStateToStaticRector.php (hunk 1 above)
Comment (simivar) [reply rules/php80/src/Rector/ClassMethod/SetStateToStaticRector.php:80]: Glad you liked it 👍

**Pull request merged**